### ISSUE TITLE
WebSocket Error [WinError 10042]

### ISSUE THREAD

**Issue by cryptoking233** [BitMEX/sample-market-maker#40]
I have the bot running all the way to the point where it connects to websocket. For some reason I am getting these errors that are preventing me from getting live data from websocket. I have entered my keys into the settings.py file and have the requirements.txt files properly installed. Not sure how to fix these errors, any help is greatly appreciated, thank you! Errors:

Importing symbol settings for XBTUSD...
Unable to find settings-XBTUSD.py.
2017-12-28 11:37:02,360 - INFO - market_maker - BitMEX Market Maker Version: v1.1
2017-12-28 09:29:43,422 - ERROR - ws_thread - [WinError 10042] An unknown, invalid, or unsupported option or level was specified in a getsockopt or setsockopt call
2017-12-28 09:29:43,423 - INFO - ws_thread - Websocket Closed
2017-12-28 09:29:44,329 - ERROR - ws_thread - Couldn't connect to WS! Exiting.

**Comment by davidjcooper** [comment]
I have the same issue but only on certain servers. I am investigating. But I believe it’s due to the version of python and the way we need to specifics the wss connection for the websocket

**Comment by cryptoking233** [comment]
What servers has it worked on for you? thanks for the help!

**Comment by cryptoking233** [comment]
I just updated my first post, there was more to the error. I think the "unable to find settings-XBTUSD.py" is causing issues. shouldn't is only be searching for settings.py?

**Comment by davidjcooper** [comment]
python 2.7.10 works for me only.

**Comment by davidjcooper** [comment]
it works for me when I use python3...

**Comment by ryanfox** [comment]
@cryptoking233 This sounds like one of your dependencies might not have installed correctly.
- What platform and python version are you running on?
- How did you download and install the market maker?
- How did you install the dependencies? Do you have the latest version of all them?

**Comment by NichJohn83** [comment]
I am also getting this error. I installed the bot just as the README instructed.

`pip install bitmex-market-maker
`
`marketmaker setup
`
Then I added my keys to the settings.py file and exited.

`marketmaker XBTUSD`

I got the error:
2018-03-08 17:29:42,758 - INFO - market_maker - BitMEX Market Maker Version: v1.1

2018-03-08 17:29:42,758 - INFO - ws_thread - Connecting to wss://testnet.bitmex.com/realtime?subscribe=quote:XBTUSD,trade:XBTUSD,instrument,order:XBTUSD,execution:XBTUSD,margin,position
2018-03-08 17:29:42,758 - INFO - ws_thread - Authenticating with API Key.
2018-03-08 17:29:42,759 - INFO - ws_thread - Started thread
